store/sqlstore/team_store.go

```diff
@@ -1504,6 +1504,32 @@ func (s SqlTeamStore) GetUserTeamIds(userId string, allowFromCache bool) ([]stri
 	return teamIds, nil
 }
 
+// GetCommonTeamIDsForUsers returns the intersection of all the teams to which the specified
+// users belong.
+func (s SqlTeamStore) GetCommonTeamIDsForUsers(userIDs []string) ([]string, error) {
+	if len(userIDs) != 2 {
+		return nil, errors.New("GetCommonTeamIDsForUsers can only be called with two users")
+	}
+	var teamIDs []string
+	const query = `
+	SELECT A.TeamId FROM (
+		SELECT TeamId FROM TeamMembers
+		WHERE UserId = :UserId0 AND DeleteAt = 0
+	) AS A INNER JOIN (
+		SELECT TeamId FROM TeamMembers
+		WHERE UserId = :UserId1 AND DeleteAt = 0
+	) AS B ON A.TeamId = B.TeamId
+	INNER JOIN Teams ON A.TeamId = Teams.Id
+	WHERE Teams.DeleteAt = 0`
+	props := map[string]interface{}{"UserId0": userIDs[0], "UserId1": userIDs[1]}
+	_, err := s.GetReplica().Select(&teamIDs, query, props)
```

Comment:
@sbishel 
In regards to the master/slave read replica lag issue, could we have a scenario where a user just joined a team that is common with another user but hasn't been replicated in the replica so potentially this query doesn't work? Something we need to be concerned with or nah?

Reply:
We don't need to worry about the use case you mentioned because those are separate actions.

Where we need to worry about this is if its all part of the same action (transaction). IOW, if we added the user to a team, then created a direct message as part of the same process or function. Then this could be an issue.

Reply:
ooooh true true true its only if its happening right after a write. Good call good call.